server/athenian/api/preloading/__main__.py

```diff
@@ -0,0 +1,110 @@
+                        help="Pushed and pulled source data that Athenian owns. DB connection "
+                             "string in SQLAlchemy format. This DB is read/write.")
+    parser.add_argument("--detailed", required=False, action="store_true",
+                        help="Whether to also show the memory usage of each column.")
+
+    return parser.parse_args()
+
+
+def main():
+    """Print memory usage for preloaded DataFrames."""
+    args = parse_args()
+    log = logging.getLogger(metadata.__package__)
+    if not check_schema_versions(args.metadata_db,
+                                 args.state_db,
+                                 args.precomputed_db,
+                                 args.persistentdata_db,
+                                 log):
+        return None
+
+    db_conns = {
+        "sdb": args.state_db,
+        "mdb": args.metadata_db,
+        "pdb": args.precomputed_db,
+        "rdb": args.persistentdata_db,
+    }
+    db_options = compose_db_options(args.metadata_db,
+                                    args.state_db,
+                                    args.precomputed_db,
+                                    args.persistentdata_db)
+
+    asyncio.run(_main(db_conns, db_options, args.detailed))
+
+
+async def _main(db_conns: dict, db_options: dict, detailed: bool):
+    tasks, dbs = [], {}
+    for db_shortcut, db_conn in db_conns.items():
+        db = ParallelDatabase(db_conn, **(db_options[f"{db_shortcut}_options"]))
+        tasks.append(db.connect())
+        dbs[db_shortcut] = db
+
+    await gather(*tasks)
+
+    mc = MemoryCachePreloader()
+    mc.preload(detailed, **dbs)
+    await mc._task
+
+    for db_name, db in dbs.items():
+        if not (cache := getattr(db, "cache", None)):
+            continue
+
+        db_total_mem = cache.memory_usage(True, True)
+        _print_summary_line(f"DB: {db_name}", db_total_mem, 1)
+        for name, df in cache.dfs.items():
+            total_mem = df.memory_usage(True, True)
+            _print_summary_line(f"DataFrame: {name}", total_mem, 2)
+            if not detailed:
+                continue
+
+            info = df._mem
+            raw, processed, percentage = info["raw"], info["processed"], info["percentage"]
+            _print_summary_line("Raw", raw["total"], 3)
+            print("--- Raw series")
+            print(raw["series"])
+
+            _print_summary_line("Processed", processed["total"], 3)
+            print("--- Processed series")
+            print(processed["series"])
+
+            _print_summary_line("Percentage", percentage["total"], 3)
+            print("--- Percentage series")
+            print(percentage["series"])
+
+
+def _print_summary_line(title, total, indent_level):
+    prefix = "-" * indent_level
+    print(f"{(prefix + ' ' + title):<35} | [total: {total}]")
+
+
+if __name__ == "__main__":
+    exit(main() is None)  # "1" for an error, "0" for a normal return
```

Comment:
Or just
```suggestion
    sys.exit(main())
```
as it converts `None` to 0.